subsys/usb/device_next/class/usbd_hid.c

```diff
@@ -231,25 +231,33 @@ static int handle_get_report(const struct device *dev,
 	const uint8_t id = HID_GET_REPORT_ID(setup->wValue);
 	struct hid_device_data *const ddata = dev->data;
 	const struct hid_device_ops *ops = ddata->ops;
+	const size_t size = net_buf_tailroom(buf);
+	int ret = 0;
 
 	switch (type) {
 	case HID_REPORT_TYPE_INPUT:
 		LOG_DBG("Get Report, Input Report ID %u", id);
-		errno = ops->get_report(dev, type, id, net_buf_tailroom(buf), buf->data);
+		ret = ops->get_report(dev, type, id, size, buf->data);
 		break;
 	case HID_REPORT_TYPE_OUTPUT:
 		LOG_DBG("Get Report, Output Report ID %u", id);
-		errno = ops->get_report(dev, type, id, net_buf_tailroom(buf), buf->data);
+		ret = ops->get_report(dev, type, id, size, buf->data);
 		break;
 	case HID_REPORT_TYPE_FEATURE:
 		LOG_DBG("Get Report, Feature Report ID %u", id);
-		errno = ops->get_report(dev, type, id, net_buf_tailroom(buf), buf->data);
+		ret = ops->get_report(dev, type, id, size, buf->data);
 		break;
 	default:
 		errno = -ENOTSUP;
 		break;
 	}
 
+	if (ret > 0) {
+		net_buf_add(buf, MIN(size, ret));
```

Comment:
Should there be an assert here to check that `ret <= size`?

Reply:
Host requesting more than report length is not what would be asserted here. When `ret <= size` is not fulfilled it means that application essentially wrote (or at least claimed to write) beyond the buffer.